src/components/themed/ExchangeQuoteComponent.tsx

```diff
@@ -22,39 +26,54 @@ interface Props {
 export const ExchangeQuote = (props: Props) => {
   const { fromTo, quote, showFeeWarning } = props
   const { request, fromNativeAmount, toNativeAmount, networkFee } = quote
-  const isFrom = fromTo === 'from'
-  const wallet = isFrom ? request.fromWallet : request.toWallet
-  const tokenId = isFrom ? request.fromTokenId : request.toTokenId
+  const { fromWallet, fromTokenId, toWallet, toTokenId } = request
 
   const theme = useTheme()
   const styles = getStyles(theme)
 
+  const isFrom = fromTo === 'from'
   const nativeAmount = isFrom ? fromNativeAmount : toNativeAmount
 
+  // Fees are assumed to be denominated in the native currency
   const feeNativeAmount = networkFee.nativeAmount
-  const feeCryptoText = useCryptoText({ wallet, tokenId, nativeAmount: feeNativeAmount, withSymbol: true })
-  const { currencyCode, denomination, isoFiatCurrencyCode } = useTokenDisplayData({
-    tokenId,
-    wallet
+  const feeCryptoText = useCryptoText({ wallet: fromWallet, nativeAmount: feeNativeAmount, withSymbol: true })
+  const {
+    currencyCode: nativeCurrencyCode,
+    denomination: nativeDenomination,
+    isoFiatCurrencyCode: nativeIsoFiatCurrencyCode
```

Comment:
`nativeIsoFiatCurrencyCode` will always be the same as `fromIsoFiatCurrencyCode` since the wallet is the same. Simplify to `isoFiatCurrencyCode`